ludwig/schema/encoders/image_encoders.py

```diff
@@ -284,368 +284,828 @@ class Stacked2DCNNEncoderConfig(BaseEncoderConfig):
-        "the output tensor is of shape (n_patches, embed_size), where n_patches is img_height x img_width "
-        "/ patch_size².",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["avg_pool"],
+
+@register_encoder_config("mobilenetv3_torch", IMAGE)
+@dataclass
+class TVMobileNetV3EncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["mobilenetv3_torch"],
+        default="mobilenetv3_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        [
+            "small",
+            "large",
+        ],
+        default="small",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVMobileNetV3Encoder"]["model_variant"],
+    )
 
-@register_encoder_config("vit", IMAGE)
-@dataclass(repr=False)
-class ViTEncoderConfig(BaseEncoderConfig):
 
+@register_encoder_config("regnet_torch", IMAGE)
+@dataclass
+class TVRegNetEncoderConfig(TVBaseEncoderConfig):
     type: str = schema_utils.StringOptions(
-        ["vit"],
-        default="vit",
+        ["regnet_torch"],
+        default="regnet_torch",
         allow_none=False,
         description="Type of encoder.",
     )
 
-    height: int = schema_utils.NonNegativeInteger(
-        default=None,
-        description="Height of the input image.",
-        parameter_metadata=ENCODER_METADATA["ViTEncoder"]["height"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        [
+            "x_1_6gf",
+            "x_16gf",
+            "x_32gf",
+            "x_3_2gf",
+            "x_400mf",
+            "x_800mf",
+            "x_8gf",
+            "y_128gf",
+            "y_16gf",
+            "y_1_6gf",
+            "y_32gf",
+            "y_3_2gf",
+            "y_400mf",
+            "y_800mf",
+            "y_8gf",
+        ],
+        default="x_1_6gf",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVRegNetEncoder"]["model_variant"],
     )
 
-    width: int = schema_utils.NonNegativeInteger(
-        default=None,
-        description="Width of the input image.",
-        parameter_metadata=ENCODER_METADATA["ViTEncoder"]["width"],
+
+@register_encoder_config("resnet_torch", IMAGE)
+@dataclass
+class TVResNetEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["resnet_torch"],
+        default="resnet_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    num_hidden_layers: int = schema_utils.PositiveInteger(
-        default=12,
-        description="Number of hidden layers in the Transformer encoder.",
-        parameter_metadata=ENCODER_METADATA["ViTEncoder"]["num_hidden_layers"],
+    model_variant: Optional[int] = schema_utils.IntegerOptions(
```

Comment:
Want to double-check if this field is really `Optional`? If so, make sure `allow_none` is `True`.

Reply:
removed optional